router.go

```diff
@@ -569,7 +579,11 @@ func (r *Router) Find(method, path string, c Context) {
 		if child := currentNode.anyChild; child != nil {
 			// If any node is found, use remaining path for paramValues
 			currentNode = child
-			paramValues[len(currentNode.pnames)-1] = search
+			if m := currentNode.findHandler(method); m != nil {
```

Comment:
I think better approach would be to compute parameter count as a field on node. Not that calling `currentNode.findHandler(method)` on each any handler is super slow but it is still unnecessary most of this work can be computed beforehand.  You can check how in `v5` branch this is done. See here for usage of [`node.paramsCount`](https://github.com/labstack/echo/blob/v5_alpha/router.go).

Reply:
Good approach, added